ncm-afsclt/src/main/perl/afsclt.pm

```diff
@@ -2,997 +2,306 @@
-      $self->warn("new AFS cache mount $new_cachemount is not a directory, ignoring");
-      $new_cachemount = $file_cachemount;
-    }
-  }
-
-  if ($new_cachemount && $file_cachemount ne $new_cachemount) {
-    $self->warn("cannot yet handle AFS cache mount move $file_cachemount -> $new_cachemount");
-  }
-
-  # Linux AFS client can autoconfigure if the AFS cache is on a
-  # separate partition (and we then won't change the other config
-  # files/run-time cache). Other machines with a separate partition
-  # will use 85% as cache size (heuristic, cache manager says 90%).
-  # Machines without a separate partition get the requested size.
-
-  # Please note that you cannot force a smaller cache if running with
-  # separate cache partition (forcing a larger one wouldn't make any sense)
-  my $dfout;
-  if (LC::Process::execute(['df', '-k', $new_cachemount],
-                           "stdout" => \$dfout,
-			   "stderr" => "stdout"
-			  ) ) {
-    my @df = split('\n', $dfout);
-
-    my @mount = grep { m{$new_cachemount} } @df;
-    if ($#mount >=0 && $mount[0] =~ m{^(.*?)\s+(\d+)\s+(\d+)\s+(\d+)\s+(\d+%)\s+(.*)}) {
-      $automatic_cache = 1;
-      my $real_cachedev = $1;
-      my $real_cachesize = $2;
-      my $real_cachemount = $6;
-      $new_cache = 0.85 * $real_cachesize;  # heuristic here
-      $self->info("found AFS cache partition on $real_cachedev, ".$real_cachesize."k");
-    } else {
-      $self->debug(2, "AFS cache $new_cachemount is not on a separate partition");
+    # LWP::Simple means  no error messages etc.
+    my $cellservdb_content = LWP::Simple::get($master_cellservdb);
+    if ( !$cellservdb_content ) {
+        $self->info("Cannot read $master_cellservdb, AFS cell info not changed");
+        return 1;
     }
-  } else {
-    $self->warn("cannot run 'df' on the AFS cache?");
-  }
-
-  # OpenAFS-1.4/Linux takes the info from /etc/sysconfig/afs
-  if( -r $sysconfig_afs) {
-    if($automatic_cache) {
-      $new_cache = "AUTOMATIC";
-    }
-    $ret &= NCM::Check::lines($sysconfig_afs,
-			    linere => '^#.*$compname-cachesize|CACHESIZE=.*',
-			    goodre => '^#.*$compname-cachesize|CACHESIZE=\s*'.quotemeta($new_cache),
-			    good   => "# changed by $compname-cachesize\nCACHESIZE=".$new_cache,
-			    keep   => 'first',
-			    add    => 'last',
-			    backup => '.old_verbose');
-
-  }
-
-  # adjust stored cache size (gets overwritten on restart for OpenAFS-1.4)
-  if ($new_cache ne "AUTOMATIC") {
-    if ($new_cache != $file_cache) {
-      if(! open (CI, '>'.$afs_cacheinfo)) {
-	$self->error("cannot open $afs_cacheinfo for writing: $!");
-	return 0;
-      }
-      print CI "/afs:$file_cachemount:$new_cache\n";
-      close(CI) or
-	$self->warn("cannot close $file_cachemount: $!");
-      $self->OK("changed AFS cache config file $afs_cacheinfo: $file_cachemount $run_cache -> $new_cache (1K blocks)");
-    } else {
-      $self->info("AFS cache size is unchanged ($new_cache K)");
+
+    if ($NoAction) {
```

Comment:
the `NoAction` is only needed for the `update_afs_cells` (or is that method alread `NoAction` safe on itself?